### PR TITLE
Gobierto Data / Including a skeleton spinner

### PR DESCRIPTION
**Pull request by jorgeatgu**
Closes https://github.com/PopulateTools/issues/issues/1079

## :v: What does this PR do?

It includes a skeleton spinner that simulates content while loading the different data. 

## :mag: How should this be manually tested?

Staging

### After this PR

after_spinner 2020-07-07 17_56_54

### CODE REVIEW

File: app/javascript/lib/vue-components/modules/SkeletonSpinner.vue [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    styleSquares() {
+      return {
+        //
+        width: ((100 - (this.squares - 1)) / this.squares) + '%',
+        height: this.heightSquare
+      }
+    },
+  },
+  methods: {
+    stylesLines() {
```
Comment (Crashillo) [inline app/javascript/lib/vue-components/modules/SkeletonSpinner.vue:57]: Why this method is not a computed one?

File: config/locales/defaults/ca.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ ca:
     - dv
     - ds
     abbr_month_names:
-    -
+    -
```
Comment (Crashillo) [inline config/locales/defaults/ca.yml:18]: Maybe could we remove these empty values?

**Review by Crashillo — APPROVED** [review]

File: app/javascript/gobierto_data/webapp/components/Dataset.vue [hunk 3]
```diff
@@ -1,111 +1,122 @@
...
   <div>
-    <div class="pure-g">
-      <div class="pure-u-1-2">
-        <h2
-          v-if="titleDataset"
-          class="gobierto-data-title-dataset"
-        >
-          {{ titleDataset }}
-        </h2>
+    <template v-if="!isDatasetLoading">
```
Comment (ferblape) [inline app/javascript/gobierto_data/webapp/components/Dataset.vue]: I think this condition is in the wrong direction: if the dataset is loading, show the spinner.  Do you mean isDatasetLoaded?

File: app/javascript/lib/vue-components/modules/SkeletonSpinner.vue (hunk 1 above)
Comment (jorgeatgu) [reply app/javascript/lib/vue-components/modules/SkeletonSpinner.vue:57]: It was. But this method determines the width of the lines, if it were computed the width would always be the same.

File: app/javascript/lib/vue-components/modules/SkeletonSpinner.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    lines: {
+      default: '0',
+      type: String,
+    }
+  },
+  computed: {
+    styleSquares() {
+      return {
+        //
```
Comment (ferblape) [inline app/javascript/lib/vue-components/modules/SkeletonSpinner.vue]: empty comment?

File: app/javascript/lib/vue-components/modules/SkeletonSpinner.vue [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        width: ((100 - (this.squares - 1)) / this.squares) + '%',
```
Comment (ferblape) [inline app/javascript/lib/vue-components/modules/SkeletonSpinner.vue]: If `this.squares` is a String shouldn't you convert it to integer or float? Careful with zero division

File: app/javascript/lib/vue-components/modules/SkeletonSpinner.vue [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    </template>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "SkeletonSpinner",
+  props: {
+    heightSquare: {
+      default: `1rem`,
```
Comment (ferblape) [inline app/javascript/lib/vue-components/modules/SkeletonSpinner.vue]: Single quotes?

**Review by ferblape — CHANGES_REQUESTED** [review]
Please review my comments

File: config/locales/defaults/ca.yml (hunk 2 above)
Comment (jorgeatgu) [reply config/locales/defaults/ca.yml:18]: This looks strange. I didn't change these files https://github.com/PopulateTools/gobierto/blob/fb23c24f82dcf53dd41ba6c65c97849f62e20f13/config/locales/defaults/ca.yml cc @ferblape @entantoencuanto

Comment (ferblape) [reply config/locales/defaults/ca.yml:18]: It looks like you added a space, but you can ignore it, we can't remove the blank here, it's a Rails default.

File: app/javascript/gobierto_data/webapp/components/Dataset.vue (hunk 3 above)
Comment (jorgeatgu) [reply app/javascript/gobierto_data/webapp/components/Dataset.vue]: You're right.

File: app/javascript/lib/vue-components/modules/SkeletonSpinner.vue (hunk 5 above)
Comment (jorgeatgu) [reply app/javascript/lib/vue-components/modules/SkeletonSpinner.vue]: In the division JS always try to transforms `String` into `Number`.

> Careful with zero division

Done

**Pull request merged**